### DISCUSSION TITLE
Audio Jack Setup Working

### DISCUSSION THREAD

**Discussion by unmanagedtn** [devmfc/debian-on-amlogic#51] [General]
The amixer Simple Control for the 3.5mm audio jack is called 'ACODEC'.

1. Edit /etc/asound.conf with

pcm.!default {
        type plug
        slave {
        pcm "hw:0,1"
        }
}

2. Run the following command lines in a shell script.

amixer sset 'FRDDR_A SINK 1 SEL' 'OUT 1' || true
amixer sset 'FRDDR_A SRC 1 EN' 'on' || true
amixer sset 'TDMOUT_B SRC SEL' 'IN 0' || true
amixer sset 'TOHDMITX I2S SRC' 'I2S B' || true
amixer sset 'TOHDMITX' 'on' || true

amixer sset 'FRDDR_B SINK 1 SEL' 'OUT 2' || true
amixer sset 'FRDDR_B SRC 1 EN' 'on' || true
amixer sset 'TDMOUT_C SRC SEL' 'IN 1' || true
amixer sset 'TOACODEC SRC' 'I2S C' || true
amixer sset 'TOACODEC OUT EN' 'on' || true
amixer sset 'TOACODEC Lane Select' '0' || true
amixer sset 'ACODEC' '255'

amixer sset 'FRDDR_C SINK 1 SEL' 'OUT 3' || true
amixer sset 'FRDDR_C SRC 1 EN' 'on' || true
amixer sset 'SPDIFOUT SRC SEL' 'IN 2' || true

alsactl store

3. reboot
4. run speaker-test

**Comment by devmfc** [comment]
Nice, thanks for sharing this!

**Reply by unmanagedtn** [reply]
trying this out on the lateset version and all i get is a dull thud from audio in a speaker-test

**Reply by unmanagedtn** [reply]
turns out that only the left audio works unless you turn it to mono in alsamixer
